src/Examine/LuceneEngine/Providers/LuceneIndex.cs

```diff
@@ -987,7 +1083,20 @@ private int ForceProcessQueueItems(bool block)
                 if (!RunAsync || block)
                 {
                     //commit the changes (this will process the deletes too)
-                    writer.Commit();
+                    if (_directory is ExamineDirectory examineDirectory)
+                    {
+                        //Calling commit causes fdt,fdx,fnm,frq,nrm,prx,tii,tis,tvd,tvf,tvx files to be deleted.
```

Comment:
This only occurs if running non-async (i.e. in tests or on shutdown). I still think that SnapshotDeletionPolicy can avoid any deletion issues for syncing